src/lib/events.ts

```diff
@@ -407,3 +407,28 @@ export function without<T extends CollectionEntry<"events">>(
 ) {
   return events.filter((event) => event.data.status !== status);
 }
+
+export async function getUpcomingEventsWithOpenCfp(limit = undefined) {
+  const upcomingEvents = await getUpcomingEvents();
+
+  const cfpEvents =
+    upcomingEvents
+      .filter(
+        (e) =>
+          e.data.cfp &&
+          (!e.data.cfp?.endDate ||
+            (e.data.cfp?.endDate &&
+              dayjs(e.data.cfp.endDate).isAfter(dayjs()))),
+      )
+      .sort(
+        (event1, event2) =>
+          (event1.data.date?.valueOf() ?? 0) -
+          (event2.data.date?.valueOf() ?? 0),
+      ) ?? [];
+
+  if (limit) {
+    return cfpEvents.slice(0, limit);
+  }
+
+  return cfpEvents;
+}
```

Comment:
Or, if you want, you can do something like this, it should work the same because `limit` is by default `undefined` and `slice` takes a second parameter as `undefined` and will return the complete list if `undefined`:


```suggestion
export async function getUpcomingEventsWithOpenCfp(limit?: number) {
  const upcomingEvents = await getUpcomingEvents();

  const cfpEvents =
    upcomingEvents
      .filter(
        (e) =>
          e.data.cfp &&
          (!e.data.cfp?.endDate ||
            (e.data.cfp?.endDate &&
              dayjs(e.data.cfp.endDate).isAfter(dayjs()))),
      )
      .sort(
        (event1, event2) =>
          (event1.data.date?.valueOf() ?? 0) -
          (event2.data.date?.valueOf() ?? 0),
      ) ?? [];

    return cfpEvents.slice(0, limit);
}
```